server/src/main/java/io/crate/auth/LoadedRSAKeyProvider.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.auth;
+
+import java.security.interfaces.RSAPrivateKey;
+import java.security.interfaces.RSAPublicKey;
+
+import com.auth0.jwt.interfaces.RSAKeyProvider;
+
+/**
+ * Key provider returning already resolved instance.
+ * Needed to avoid double access of the JWK endpoint.
+ * First time we access it for building Algorithm object - as cannot rely on optional and also unsafe token header "alg".
+ * Second access might happen on Algorithm.verify() call but if Algorithm has this class as provider, key is taken from already resolved data.
+ */
```

Comment:
I don't understand these comments. Afaik this class is just a simple container for the resolved public key. It's not cached anywhere. It will be created on each authentication call. These comments do not add value (at least for me) but instead confuses.

Reply:
What I wanted to say that it helps to avoid 2 API calls per `authenticate`.
first - to resolve Algorithm, by getting a JWK object (which contains both public key and alg)
second call happens implicitly inside library's `algorithm.verify`.

Reply:
> It's just a container but we then can use this container in library API to avoid some internal redundant calls to the endpoint.

Isn't this the whole point of implementing `RSAKeyProvider`? If so, this is API behaviour which doesn't need documentation in our implementation. I'd just drop all this.